packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -751,13 +753,21 @@ class PageableListState<T> extends ScrollableListState<T, PageableList<T>> {
     return scrollTo(newScrollOffset, duration: config.duration, curve: config.curve).then(_notifyPageChanged);
   }
 
-  int get currentPage => (scrollOffset / config.itemExtent).floor() % itemCount;
+  int get currentPage => _currentPage;
 
   void _notifyPageChanged(_) {
+    _currentPage = (scrollOffset / config.itemExtent).floor() % itemCount;
     if (config.onPageChanged != null)
       config.onPageChanged(currentPage);
   }
 
+  void _updateScrollBehavior() {
```

Comment:
why not in `_handleSizeChanged` ? `_updateScrollBehavior` is called from build, which doesn't seem like a good time to be calling scrollTo().